modules/analysis/rosbag-image-pipeline-sfn/stack.py

```diff
@@ -0,0 +1,623 @@
+                            },
+                        },
+                    ]
+                },
+            },
+        )
+
+        obj_detection_map_task = sfn.Map(
+            self,
+            "Object Detection Parallel Map",
+            items_path="$.ImageDirs.S3Paths",
+            max_concurrency=object_detection_job_concurrency,
+        ).item_processor(
+            self.processing_job_add_wait(
+                id=object_detection_task.id,
+                start_process_task=object_detection_task,
+                wait_time_seconds=15,
+            )
+        )
+
+        lane_detection_map_task = sfn.Map(
+            self,
+            "Lane Detection Parallel Map",
+            items_path="$.ImageDirs.S3Paths",
+            max_concurrency=lane_detection_job_concurrency,
+        ).item_processor(
+            self.processing_job_add_wait(
+                id=lane_detection_task.id,
+                start_process_task=lane_detection_task,
+                wait_time_seconds=15,
+            ),
+        )
+
+        emr_task_chain = self.create_emr_task_chain(
+            emr_app_id=emr_app_id,
+            emr_job_exec_role_arn=emr_job_exec_role_arn,
+            target_bucket=target_bucket,
+            artifacts_bucket=artifacts_bucket,
+            logs_bucket=logs_bucket,
+            image_topics=image_topics,
+            tracking_table=tracking_table,
+            detection_ddb_table=detection_ddb_table,
+            batch_id=sfn_batch_id,
+        )
+
+        # Define state machine
+        definition = (
+            create_batch_task.next(
+                sfn.Parallel(self, "Sensor Extraction", result_path=sfn.JsonPath.DISCARD)
+                .branch(image_extraction_step_machine_task)
+                .branch(parquet_extraction_step_machine_task)
+            )
+            .next(get_image_dirs_task)
+            .next(
+                sfn.Parallel(self, "Image Labelling", result_path=sfn.JsonPath.DISCARD)
+                .branch(obj_detection_map_task)
+                .branch(lane_detection_map_task)
+            )
+            .next(emr_task_chain)
+        )
+
+        self.state_machine = sfn.StateMachine(
+            self,
+            "StateMachine",
+            state_machine_name=f"{project_name}-{deployment_name}-rosbag-image-pipeline-{hash}",
+            definition_body=sfn.DefinitionBody.from_chainable(definition),
+        )
+
+        NagSuppressions.add_stack_suppressions(
+            self,
+            apply_to_nested_stacks=True,
+            suppressions=[
+                NagPackSuppression(
+                    id="AwsSolutions-IAM4",
+                    reason="Managed Policies are for service account roles only",
+                ),
+                NagPackSuppression(
+                    id="AwsSolutions-IAM5",
+                    reason="Resource access restriced to resources",
+                ),
+            ],
+        )
```

Comment:
I wonder if we need these nag suppressions. Also they don't do anything unless we also add:
```python
cdk.Aspects.of(self).add(cdk_nag.AwsSolutionsChecks(log_ignores=True))
```

Reply:
I don't think we need either